internal/gatewayapi/translator.go

```diff
@@ -0,0 +1,515 @@
+				Name:    irListenerName(listener),
+				Address: "0.0.0.0",
+				Port:    uint32(listener.Port),
+			}
+			if listener.Hostname != nil {
+				irListener.Hostnames = append(irListener.Hostnames, string(*listener.Hostname))
+			}
+			xdsIR.HTTP = append(xdsIR.HTTP, irListener)
+		}
+	}
+}
+
+func (t *Translator) ProcessHTTPRoutes(httpRoutes []*v1beta1.HTTPRoute, gateways []*GatewayContext, cache Cache, xdsIR *ir.Xds) []*HTTPRouteContext {
+	var relevantHTTPRoutes []*HTTPRouteContext
+
+	for _, h := range httpRoutes {
+		httpRoute := &HTTPRouteContext{HTTPRoute: h}
+
+		// Find out if this route attaches to one of our Gateway's listeners,
+		// and if so, get the list of listeners that allow it to attach for each
+		// parentRef.
+		var relevantRoute bool
+		for _, parentRef := range httpRoute.Spec.ParentRefs {
+			isRelevantParentRef, selectedListeners := GetReferencedListeners(parentRef, gateways)
+
+			// Parent ref is not to a Gateway that we control: skip it
+			if !isRelevantParentRef {
+				continue
+			}
+			relevantRoute = true
+
+			parentRefCtx := httpRoute.GetRouteParentContext(parentRef)
+
+			if !HasReadyListener(selectedListeners) {
+				parentRefCtx.SetCondition(v1beta1.RouteConditionAccepted, metav1.ConditionFalse, "NoReadyListeners", "There are no ready listeners for this parent ref")
+				continue
+			}
+
+			var allowedListeners []*ListenerContext
+			for _, listener := range selectedListeners {
+				if listener.AllowsKind(v1beta1.RouteGroupKind{Group: GroupPtr(v1beta1.GroupName), Kind: KindHTTPRoute}) && listener.AllowsNamespace(cache.GetNamespace(httpRoute.Namespace)) {
+					allowedListeners = append(allowedListeners, listener)
+				}
+			}
+
+			if len(allowedListeners) == 0 {
+				parentRefCtx.SetCondition(v1beta1.RouteConditionAccepted, metav1.ConditionFalse, v1beta1.RouteReasonNotAllowedByListeners, "No listeners included by this parent ref allowed this attachment.")
+				continue
+			}
+
+			parentRefCtx.SetListeners(allowedListeners...)
+
+			parentRefCtx.SetCondition(v1beta1.RouteConditionAccepted, metav1.ConditionTrue, v1beta1.RouteReasonAccepted, "Route is accepted")
+		}
+
+		if !relevantRoute {
+			continue
+		}
+
+		relevantHTTPRoutes = append(relevantHTTPRoutes, httpRoute)
+
+		for _, parentRef := range httpRoute.parentRefs {
+			// Skip parent refs that did not accept the route
+			if !parentRef.IsAccepted() {
+				continue
+			}
+
+			// Need to compute Route rules within the parentRef loop because
+			// any conditions that come out of it have to go on each RouteParentStatus,
+			// not on the Route as a whole.
+			var routeRoutes []*ir.HTTPRoute
+
+			// compute matches, filters, backends
+			for _, rule := range httpRoute.Spec.Rules {
+				var ruleRoutes []*ir.HTTPRoute
+
+				// A rule is matched if any one of its matches
+				// is satisfied (i.e. a logical "OR"), so generate
+				// a unique IR HTTPRoute per match.
+				for _, match := range rule.Matches {
+					irRoute := &ir.HTTPRoute{}
+
```

Comment:
`HTTPRoute` expects `Name` to be set, this might be a good time for me/someone to pick up https://github.com/envoyproxy/gateway/issues/119

Reply:
ah yeah good catch. we'll have to come up with a naming scheme - in Contour we do some hashing of some of the route properties to come up with a unique name, maybe something similar here?

Reply:
I think something as simple as what you have for the listener should be fine for now

Reply:
WFM, can always change later.

Reply:
Added names in https://github.com/envoyproxy/gateway/pull/155/commits/4bf9c657fcd4786a890e954d6e1daf786f7b0f7e, it's a little goofy given that 1 Gateway API HTTPRoute can translate to many IR HTTPRoutes and we probably want a unique name for each, but I think should work as an initial impl.